src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+    public static void delete() {
+        int index2 = scan.nextInt();
+        Task goneTask = list.get(index2 - 1);
+        list.remove(goneTask);
+
+        System.out.println("Alright, I have removed the following task from your list:\n" + goneTask);
+        System.out.println("Now you have " + list.size() + " task(s) in the list.");
+    }
+
+    /**
+     * Saves the user's list as a text file in the hard disk.
+     * @param fileName the file name of the file to be saved.
+     * @throws IOException if file cannot be created.
+     */
+    public static void save(String fileName) throws IOException {
+        File list = new File(fileName);
+
+        FileWriter saver = new FileWriter(list);
+
+        for (int i = 0; i < Duke.list.size(); i++) {
+            saver.write(Duke.list.get(i).toString() + System.lineSeparator());
+        }
+
+        System.out.println("I have saved your list for your future reference. :D");
+        saver.close();
+    }
+
+    /**
+     * Retrieves the saved list file from the hard disk.
+     * @param fileName the file name of the file to be loaded.
+     * @return an array list of tasks that the user previously added.
+     * @throws FileNotFoundException if file does not exist yet.
+     */
+    public static ArrayList<Task> get(String fileName) throws FileNotFoundException {
+        File list = new File(fileName);
+        Scanner scanner = new Scanner(list);
+
+        while (scanner.hasNext()) {
+            String fullString = scanner.nextLine();
+            if (fullString.contains("[T]")) {
+                String description = fullString.substring(7);
+
+                Task nextTask = new Todo(description);
+                Duke.list.add(nextTask);
+
+            } else if (fullString.contains("[D]")) {
+                int descriptionEnd = fullString.indexOf('(');
+                int deadlineEnd = fullString.indexOf(')');
+
+                String description = fullString.substring(7, descriptionEnd - 1);
+                String deadline = fullString.substring(descriptionEnd + 1, deadlineEnd);
+
+                Task nextTask = new Deadline(description, deadline);
+                Duke.list.add(nextTask);
+
+            } else if (fullString.contains("[E]")) {
+                int descriptionEnd = fullString.indexOf('(');
+                int startTimeEnd = fullString.indexOf("to");
+                int endTimeEnd = fullString.indexOf(')');
+
+                String description = fullString.substring(7, descriptionEnd - 2);
+                String startTime = fullString.substring(descriptionEnd + 1, startTimeEnd - 1);
+                String endTime = fullString.substring(startTimeEnd - 1, endTimeEnd);
+
+                Task nextTask = new Event(description, startTime, endTime);
+                Duke.list.add(nextTask);
+            }
+        }
+
+        return Duke.list;
+    }
+
+    /**
+     * Says goodbye to Mitsuki and exits the Chat bot.
+     */
+    public static void bye() {
```

Comment:
Following our coding standards, method names must be verbs. Perhaps `showBye()` would be more intuitive? This issue applies for some of the method names above in this file as well.